pkg/cloud/vsphere/provisioner/govmomi/create.go

```diff
@@ -499,13 +499,18 @@ func (pv *Provisioner) getCloudProviderConfig(cluster *clusterv1.Cluster, machin
 		klog.Infof("Using input vSphere server url: %s", server)
 	}
 
+	username, password, err := pv.GetVsphereCredentials(cluster)
+	if err != nil {
+		return "", err
```

Comment:
Looking at the `GetVsphereCredentials()` method, the mechanism around getting the credentials is so that if the VsphereCredentialSecret name is defined in the ClusterConfig we ignore the username and password fields completely and fail with an error if we are not able to read the credentials from the secret.

I think it might make sense to have this behave the same way as when getting the credentials to create govmomi sessions.

Reply:
Ah, I did not look at t GetVsphereCredentials to see what it was doing, and I see it's doing what I was requesting.